src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -87,6 +89,48 @@ public void setSoTimeout(int soTimeout) {
     this.soTimeout = soTimeout;
   }
 
+  public void setListeners(List<JedisCommandListener> listeners) {
+    if (listeners == null) {
+      if (!this.listeners.isEmpty()) { // No reason to re-create an already empty list
+        this.listeners = new LinkedList<>();
+      }
+    } else {
+      this.listeners = listeners;
```

Comment:
that might be an issue and create a way for users to change the list without calling addListener/removeListener. I think listeners should be final and the listeners should be copied to the list.

Reply:
OK, I've renamed this method `addListeners` and updated the behavior to just add new listeners, instead of resetting the list.